src/source-updater.js

```diff
@@ -294,13 +295,34 @@ export default class SourceUpdater extends videojs.EventTarget {
       };
     }
 
+    if (type === 'audio' && this.videoBuffer && !this.videoAppendQueued) {
+      this.delayedAudioAppendQueue_.push([{type, bytes, videoSegmentTimingInfoCallback}, doneFn]);
+      this.logger_(`delayed audio append of ${bytes.length} until video append`);
+      return;
+    }
+
     pushQueue({
       type,
       sourceUpdater: this,
       action,
       doneFn,
       name: 'appendBuffer'
     });
+
+    if (type === 'video') {
+      this.videoAppendQueued = true;
```

Comment:
we should default this to `false` in the constructor as well.

Does it make sense to reset this value anywhere beyond dispose?

Reply:
I will add it to the constructor. I don't think we need to set it anywhere else because we should always be disposing the source updater and creating another when we need to work with different/new source buffers.